### PR TITLE
[3.4] Fixed everything found by the spellcheck builder

### PR DESCRIPTION
**Pull request by wouterj**
This completes what I started in #12610 

I've also committed the custom word list and Sphinx configuration. This way, everyone is able to run it (and I accidentially removed the word list after the last time, so I don't want to loose it again).

Some quick notes:

* Many titles that consisted of option or function names are transformed to be a literal. This is supported by Sphinx and all references still work. On symfony.com code elements in a title are ignored and titles are displayed as normal (see https://symfony.com/doc/current/components/filesystem for an example)
* I once again updated many cases of library names. I used the main website/documentation of each library to find the proper casing (e.g. NGINX, webpack, npm and HInclude.js).

*To avoid merge conflicts, I will merge this PR asap. Don't worry about the merge conflicts: I'll resolve them once and then they should no longer occur*

### CODE REVIEW

File: templating/hinclude.rst [hunk 1]
```diff
@@ -1,10 +1,10 @@
 .. index::
     single: Templating; hinclude.js
 
-How to Embed Asynchronous Content with hinclude.js
+How to Embed Asynchronous Content with HInclude.js
 ==================================================
 
-Controllers can be embedded asynchronously using the `hinclude.js`_ JavaScript library.
+Controllers can be embedded asynchronously using the `HInclude.js`_ JavaScript library.
```
Comment (javiereguiluz) [inline templating/hinclude.rst]: Super minor but ... according to http://mnot.github.io/hinclude/, this is called `HInclude` and `hinclude.js` is only used to refer to the specific JS file you need to load in your pages.

**Review by javiereguiluz — APPROVED** [review]
Wouter, thanks a lot for this huge effort!

File: templating/hinclude.rst (hunk 1 above)
Comment (wouterj) [reply templating/hinclude.rst]: Wow, great find. You're completely correct, I've fixed it (also in the http cache and reference articles)

**Comment by wouterj** [comment]
@javiereguiluz this PR introduces some usage of the `.. spelling::` directive that is provided by the `sphinxcontrib-spelling` package python package. Using it should be as simple as installing this package using pip. Are we able to install this package on the symfony.com server? (requires `apt install enchant` and `pip install sphinxcontrib-spelling==4.2.0 pyenchant`, we have to have v4.2.0 of the spelling package).

An alternative solution would be to remove this directives and add the remaining spellings to the wordlist and comment out the spelling configuration. But I'm not a huge fan of that (i.e. `plaintext` is correct in Security concept, but wrong in e.g. Console documentation).

File: reference/forms/types/textarea.rst [hunk 2]
```diff
@@ -30,7 +30,7 @@ Renders a ``textarea`` HTML element.
 .. tip::
 
     If you prefer to use an **advanced WYSIWYG editor** instead of a plain
-    textarea, consider using the FOSCKEditorBundle community bundle. Read
+    text area, consider using the FOSCKEditorBundle community bundle. Read
```
Comment (xabbuh) [inline reference/forms/types/textarea.rst]: Maybe use `textarea` instead?

File: doctrine/registration_form.rst [hunk 3]
```diff
@@ -3,6 +3,10 @@
    single: Form; Simple Registration Form
    single: Security; Simple Registration Form
 
+.. spelling::
+
+    plaintext
```
[the remark this answers is not in the document]
Comment (wouterj) [reply doctrine/registration_form.rst]: Plaintext is a fun one. It's "plaintext" when you talk about encrypted text (so not-encrypted text is called "plaintext"). When you talk about plain text in any other case (such as not decorated text), it's called "plain text".

File: bundles/best_practices.rst [hunk 4]
```diff
@@ -188,7 +188,7 @@ The index file (for example ``Resources/doc/index.rst`` or
 ``Resources/doc/index.md``) is the only mandatory file and must be the entry
 point for the documentation. The
 :doc:`reStructuredText (rST) </contributing/documentation/format>` is the format
-used to render the documentation on symfony.com.
+used to render the documentation on ``symfony.com``.
```
[the remark this answers is not in the document]
Comment (wouterj) [reply bundles/best_practices.rst]: I'm not sure tbh. It's the only occurence of lowercase "symfony" that makes sense, so I was a bit lazy and transformed it into literal text.

File: components/dotenv.rst [hunk 5]
```diff
@@ -28,7 +28,7 @@ Sensitive information and environment-dependent settings should be defined as
 environment variables (as recommended for `twelve-factor applications`_). Using
 a ``.env`` file to store those environment variables eases development and CI
 management by keeping them in one "standard" place and agnostic of the
-technology stack you are using (Nginx vs PHP built-in server for instance).
+technology stack you are using (NGINX vs PHP built-in server for instance).
```
[the remark this answers is not in the document]
Comment (wouterj) [reply components/dotenv.rst]: Hmm, except from the top bar:

image

And everything that can be discovered on this page: (+ footer) https://www.nginx.com/resources/library/complete-nginx-cookbook/

But then, https://nginx.org/en/docs/ seems to use lowercase nginx again... :confused:

File: bundles/best_practices.rst (hunk 4 above)
[the remark this answers is not in the document]
Comment (javiereguiluz) [reply bundles/best_practices.rst]: Maybe:

```
``symfony.com`` -> Symfony website.
```

File: reference/forms/types/textarea.rst (hunk 2 above)
Comment (javiereguiluz) [reply reference/forms/types/textarea.rst]: I'd even go for this to be more explicit:

```
``<textarea>``
```

File: components/dotenv.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (javiereguiluz) [reply components/dotenv.rst]: Apparently, you can spell `nginx` in any way you want 🙈

image

**Comment by javiereguiluz** [comment]
@wouterj about the `.. spelling` directive ... I don't know what to say yet. I'm not sure if we could install this package ... and I don't know if this could impact @weaverryan's new build tool.

**Comment by wouterj** [comment]
For the new build tool, it would be a matter of creating the PHP class for this directive. Nothing needs to be rendered, so we just need to have the class to not have the tool fail on not knowing the directive.

However, the easiest is to just add them to the global word list. I'll make that chance shortly :)

File: components/dotenv.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (wouterj) [reply components/dotenv.rst]: Alright, I've changed NGINX to "nginx" everywhere. This is consistent with the official nginx docs and I have the feeling it disrupts the readflow less than an all-uppercase word.

**Pull request merged**